.github/workflows/ci_tests.yml

```diff
@@ -43,6 +43,23 @@ jobs:
         echo $id
         if [[ $id != null ]]; then "true" ; else "false"; fi
 
+  # PR must be labeled "ready to be merged" to run CUDA jobs.
+  # This job will fail if this is not the case.
+  # Note: Label should only be added if PR is in merge queue.
+  check_ready_to_be_merged:
+    needs: [check_pull_request_is_not_a_draft]
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Check that the PR is ready to be merged
+      run: |
+        labels=$(curl -H "Accept: application/vnd.github+json" \
+           https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.number }} \
+           | jq '.labels')
+        ready_label="ready to be merged"
+        echo "Is PR Ready to be Merged? (Check for $ready_label label)"
+        echo "Labels found are: $labels"
+        if [[ $labels == *"$ready_label"* ]]; then "true" ; else "false"; fi
```

Comment:
We were condensing our request as 
```
    exit $(curl -sS -H "Travis-API-Version: 3" \
           -X GET https://api.travis-ci.com/build/$TRAVIS_BUILD_ID/jobs | \
           jq '[ .jobs[] | select( (.stage.name == "non_blocking_failures") and (.allow_failure == true) and (.state != "passed")) ] | length')
```
Do you think we could use the same pattern?

Reply:
Also, what do you think of checking that the PR's `draft` the same way?

Reply:
> > Do you think we could use the same pattern?
> 
> I don't think so, as I am using the Github API instead of Travis's.

Couldn't we get the `draft` status for [this service](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request)?